### PR TITLE
docs: suggest to let Renovate bypass mandatory Pull Request reviews

### LINKED ISSUE #15659: Renovate bot can be excepted from branch protection rules on GitHub
### Describe the proposed change(s).

You can now "consistently allow GitHub Apps as exceptions to branch protection rules". [^gh-blog]

We should document this new feature somewhere appropriate in our docs. It may be relevant to people who use `automerge=branch` for example? [^branch-automerge-docs]

[^gh-blog]: https://github.blog/changelog/2022-05-17-consistently-allow-github-apps-as-exceptions-to-branch-protection-rules/
[^branch-automerge-docs]: https://docs.renovatebot.com/noise-reduction/#branch-automerging)

### PR DESCRIPTION
**Pull request by Shegox**
## Changes

This PR recommends that users should add Renovate to the list of users, which are allowed to bypass mandatory Pull Request instead of installing the auto-approve App. 
This has the advantage that no additional App is required. In the long term, renovate-approve could be deprecated in favor of this approach overall.

## Context

GitHub added this new feature in November 2021 and I couldn't find it yet in the Renovate docs.

Fixes #15659

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

Merging without approval: https://github.com/tobias-something-something/automerge-test/pull/3
image
although required PRs are configured
image

### CODE REVIEW

**Comment by HonkingGoose** [comment]
> GitHub added this new feature in November 2021 and I couldn't find it yet in the Renovate docs.

You're totally right! 😉 I created an issue to figure out how to describe the branch protection exception for Renovate in our docs:

- https://github.com/renovatebot/renovate/issues/15659

The problem is that we should probably change more than just add a single line somewhere in the docs. And also, if we tell our users to add Renovate bot as an exception, it should work for `automerge=pr` and `automerge=branch`.

**Comment by Shegox** [comment]
> > GitHub added this new feature in November 2021 and I couldn't find it yet in the Renovate docs.
> 
> You're totally right! 😉 I created an issue to figure out how to describe the branch protection exception for Renovate in our docs:
> 
> * [Renovate bot can be excepted from branch protection rules on GitHub #15659](https://github.com/renovatebot/renovate/issues/15659)
> 
> The problem is that we should probably change more than just add a single line somewhere in the docs. And also, if we tell our users to add Renovate bot as an exception, it should work for `automerge=pr` and `automerge=branch`.

I searched if there was an issue already open for this, but missed that one. 😅 

In my test, a PR merge worked.

File: docs/usage/key-concepts/automerge.md
Comment (rarkins) [reply docs/usage/key-concepts/automerge.md:170]: Good to mention both, seeing as the GitHub feature is new and we're not 100% sure it's superior in every way

**Comment by Shegox** [comment]
And testing using `autoMergeType: branch` via https://github.com/tobias-something-something/automerge-test/commit/b5df255050fc972f95cf42714e1c7c5584d6e42c, works as well as expected.

So overall it seems to work for all my tested cases: branch and PR and even works, if "Require review from Code Owners" is enabled: 
image

**Review by rarkins — DISMISSED** [review]

File: docs/usage/key-concepts/automerge.md
Comment (HonkingGoose) [inline docs/usage/key-concepts/automerge.md:169]:
```suggestion
If you're on `github.com` or GitHub Enterprise Server (`>=3.4`) you can let Renovate bypass the mandatory Pull Request reviews using the "[Allow specified actors to bypass required pull requests](https://github.blog/changelog/2021-11-19-allow-bypassing-required-pull-requests/)" option in your branch protection rules.
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
Small style change.

**Review by HonkingGoose — APPROVED** [review]

**Pull request merged**
